src/mirror.c

```diff
@@ -206,46 +213,85 @@ static enum swupd_code set_mirror_url(char *url)
 	return ret;
 }
 
+bool mirror_is_set(void)
+{
+	bool mirror_set;
+	char *version_path;
+	char *content_path;
+
+	version_path = mk_full_filename(path_prefix, MIRROR_VERSION_URL_PATH);
+	content_path = mk_full_filename(path_prefix, MIRROR_CONTENT_URL_PATH);
+	mirror_set = file_exists(version_path) && file_exists(content_path);
+
+	free_string(&version_path);
+	free_string(&content_path);
+
+	return mirror_set;
+}
+
 void handle_mirror_if_stale(void)
 {
 	char *ret_str = NULL;
 	char *fullpath = NULL;
 
+	/* make sure there is a mirror set up first */
+	if (!mirror_is_set()) {
+		return;
+	}
+
+	info("Checking mirror status\n");
+
 	fullpath = mk_full_filename(path_prefix, DEFAULT_VERSION_URL_PATH);
 	int ret = get_value_from_path(&ret_str, fullpath, true);
 	if (ret != 0 || ret_str == NULL) {
 		/* no versionurl file here, might not exist under --path argument */
 		goto out;
 	}
+
+	/* before trying to get the latest version let's make sure the central version is up */
+	if (!content_url_is_local && !check_connection(NULL, ret_str)) {
```

Comment:
check_connection returns an error int so the correct is
check_connection() != 0.

Reply:
done